TFEngine.py

```diff
@@ -554,6 +555,8 @@ def __init__(self, config=None):
     self.use_search_flag = config.value("task", None) == "search"
     self.use_eval_flag = config.value("task", None) != "forward"
     self._const_cache = {}  # type: dict[str,tf.Tensor]
+    self.source_voc = None
+    self.target_voc = None
```

Comment:
That should not be part of the `Engine` class. But see my comment below.

Reply:
See my comments here: https://github.com/rwth-i6/returnn/pull/48/files/c0aa865b600430eae3d272d022def324f1964107..d5cc0cafdc3de214e0c7ef71834bf7f11d022097